docs/training/training-accelerate-dep.md

```diff
@@ -0,0 +1,114 @@
+
+### Package Management
+This inclusion requires one additional python package to be managed as a dependency. Currently, that package is `accelerate==0.34.2` but we plan to immediately upgrade to `accelerate==1.0.0` once it is moved from pre-release to official release.
+
+
+## Long-Term Bonuses and Risks
+The two clearest long-term bonuses are:
+ - Simplifies the process of onboarding additional sharding frameworks in the future, as well as deprecating existing sharding frameworks
+ - Vastly improves the readability and maintenance of our code by avoiding diverging paths for various sharding frameworks
+
+There are, however, important risks to be considered. Ultimately, if we are confident that we wish to eventually stick with a single sharding framework in the future and contribute directly to that project, the risks of including Accelerate may outweigh the benefits. Please read the next section carefully to understand why, if Accelerate no longer provides a significant benefit to us, should be removed.
+
+
+### Why Stop There?
+With the inclusion of a Hugging Face abstraction library, this begs the question "why not keep going and pull in more of the HF stack? Won't it simplify things further then?"
+
+It is important to reiterate here the clear downsides of HF Accelerate, and what we wish to avoid moving forward. Accelerate makes our code more opaque, and hinders our ability to easily understand and customize as needed without being reliant upon a third-party dependency. We become reliant upon HF bug fixes, release cycles, and documentation, and without an explicit member of the HF community to manage these for us, this introduces risk that has to be heavily considered.
+
+For example, if we wanted to change how model saving worked, or a user wanted to simply understand how model saving worked, they would no longer be able to do so through our library directly, but instead would need to read HF documentation, and contribute back to HF repos with the hope that the customization is considered generally useful and approved. More niche or experimental customizations may not be approved at all, and would have to be patched in directly or overwritten in a rather hacky manner within our library. This as described is not the most dire of circumstances, as one could argue that if it is not ready to be included upstream, it may not be ready to be included in our library. Where things get more serious, however, is with the introduction of bugs. At this very moment, there is a bug in `v0.34.2` that means FSDP model saving does not behave as expected. This fix is resolved in `v1.0.0rc1`, but there is no official release yet that can be pulled in with this fix. So for RHEL AI 1.2, we are currently maintaining a patch directly in our library to fix the bug while working with `v0.34.2`, and it will sit there temporarily until Accelerate has a new official release.
```

Comment:
Isn't this inherent to all usage of a library?

Reply:
Not necessarily, this is the inclusion of a library that sits directly on top of and obfuscates the internal implementation that was previously exposed directly within our library. So it is inherent to any library being pulled in as an abstraction layer for previously directly-implemented code, but a new library/dependency can be pulled in for other reasons like additional functionality, separate code paths, etc. That being said, you are correct that this is a very widely-applicable and usually assumed claim, but it had to be re-iterated for the people that were curious as to why we didn't just pull in ALL HF libraries and try to abstract away everything 😅 (you'd be surprised how often we've received this question lol)